src/examples/fake_device_demo.rb

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env ruby
+
+$LOAD_PATH.unshift(File.expand_path('../../lib', __FILE__))
+
+require "storage/fake_probing.rb"
+require "storage/fake_device_factory.rb"
+
+FILENAME = "fake-devicegraphs"
+
+fake_probing = Yast::Storage::FakeProbing.new
+devicegraph = fake_probing.devicegraph
+factory = Yast::Storage::FakeDeviceFactory.new(devicegraph)
+factory.load_yaml_file("#{FILENAME}.yml")
+
+fake_probing.to_probed
+puts("Disks:")
+probed = Yast::Storage::StorageManager.instance.probed
+fake_probing.dump_disks(probed)
```

Comment:
I disagree.

Why would we want to pollute the StorageManager with something that absolutely does not belong there? It provides infrastructure for using the new libstorage. It is a fringe case that anybody might need or want to use a different Storage::Environment for different purposes, one of which is faking those device trees. But it's not limited to that, it's very special, and it's pretty far away from its normal usage. That's why there is this special class FakeProbing. It uses the StorageManager for its special setup, but it does not extend the very visible public API of the StorageManager. This is on purpose.

This is also why the FakeDeviceFactory gets the device graph as an input parameter rather than returning a newly created one. FakeProbing sets up the special case, FakeDeviceFactory uses it - it creates fake devices on the device graph it gets. It is a factory to create fake devices, not fake devicegraphs.